plugins/sql-language/src/main/java/org/polypheny/db/sql/SqlLanguagePlugin.java

```diff
@@ -216,6 +221,32 @@ public static void startup() {
     }
 
 
+    private static QueryContext removeLimit( QueryContext queryContext ) {
+        String lowercase = queryContext.getQuery().toLowerCase();
+        if ( !lowercase.contains( LIMIT ) ) {
+            return queryContext;
+        }
+
+        // ends with "LIMIT <number>" or "LIMIT <number>;" with optional whitespace, matches <number>
```

Comment:
Polypheny supports OFFSET; thus LIMIT may not be the last part of the query. 
```
select * from emps limit 1 offset 1
```

Reply:
Even worse Polypheny also supports:
`SELECT * FROM emps LIMIT 1,1`
Which also was never considered, added the correct regex.
Maybe we should think about removing this method all together as it "destroys" performance for all queries, just for some convenience in the frontend...

Reply:
Yes... I think the only somehow viable solution would be to work with the parsed query instead of trying to preparse this. The longer I look at this method, the more issues I see...

Also the name is confusing, isnt it? It does not *remove* the limit, only extracts it. The query itself is not changed. Why are we doing this anyway?

Reply:
In the console view of the UI we automatically batch the result of a query into chunks of ten tuples, except if the user specifies a limit. This was initially added as a more streamlined and error prone way of extracting the limit for this (initially this was just a string check). 
Question is, if we should even support this, as I question the importance of the use-use case in the first place.